batch/batches/Delete/Engines/KDeletingEntryEngine.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * @package Scheduler
+ * @subpackage Delete
+ */
+class KDeletingEntryEngine extends KDeletingEngine
+{
+	/* (non-PHPdoc)
+	 * @see KDeletingEngine::delete()
+	 */
+	protected function delete(KalturaFilter $filter)
+	{
+		return $this->deleteEntries($filter);
+	}
+	
+	/**
+	 * @param KalturaBaseEntryFilter $filter The filter should return the list of entries that need to be deleted
+	 * @return int the number of deleted entries
+	 */
+	protected function deleteEntries(KalturaBaseEntryFilter $filter)
+	{
+		if (!$filter->orderBy)
+		{
+			$filter->orderBy = KalturaBaseEntryOrderBy::UPDATED_AT_ASC;
+		}
+		
+		$entriesList = KBatchBase::$kClient->baseEntry->listAction($filter, $this->pager);
+		if (!$entriesList->objects || !count($entriesList->objects))
+		{
+			return 0;
+		}
+		
+		$entriesToDeletePerRequest = KBatchBase::$taskConfig->params->entriesToDeletePerRequest;
+		$waitBetweenRequestsInSeconds = KBatchBase::$taskConfig->params->waitBetweenRequestsInSeconds;
+		
+		$entriesCount = 0;
+		$countResults = 0;
+		while ($entriesCount < count($entriesList->objects))
+		{
+			KBatchBase::$kClient->startMultiRequest();
+			for ($i = $entriesCount; $i < min($entriesCount + $entriesToDeletePerRequest, count($entriesList->objects)); $i++)
```

Comment:
this is not fully clear and seems like it will work only from one page results of base entry list
maybe worth simplifying it and just do paginations on 30 page size filter multiple times and create multi request for each one of them